csharp/src/Ice/LoggerAdmin.cs

```diff
@@ -66,44 +138,25 @@ public void
 
             if (_traceLevel > 0)
             {
-                _logger.Trace(TraceCategory, "attached `" + remoteLogger.ToString() + "'");
+                _logger.Trace(TraceCategory, $"attached `{remoteLogger}'");
             }
 
             if (initLogMessages.Count > 0)
             {
-                FilterLogMessages(initLogMessages, filters.MessageTypes, filters.TraceCategories, messageMax);
+                FilterLogMessages(initLogMessages, messageTypes, traceCategories, messageMax);
             }
 
-            try
+            remoteLogger.Queue(async prx =>
             {
-                remoteLogger.InitAsync(_logger.GetPrefix(), initLogMessages.ToArray()).ContinueWith(
-                    (t) =>
-                    {
-                        try
-                        {
-                            t.Wait();
-                            if (_traceLevel > 1)
-                            {
-                                _logger.Trace(TraceCategory, "init on `" + remoteLogger.ToString()
-                                              + "' completed successfully");
-                            }
-                        }
-                        catch (AggregateException ae)
-                        {
-                            DeadRemoteLogger(remoteLogger, _logger, ae.InnerException!, "init");
-                        }
-                    },
-                    System.Threading.Tasks.TaskScheduler.Current);
-            }
-            catch (System.Exception ex)
-            {
-                DeadRemoteLogger(remoteLogger, _logger, ex, "init");
-                throw;
-            }
+                await prx.InitAsync(_logger.GetPrefix(), initLogMessages.ToArray());
+                if (_traceLevel > 1)
+                {
+                    _logger.Trace(TraceCategory, $"init on `{remoteLogger}' completed successfully");
```

Comment:
Since we have no idea if it completes or not, I think you should change to message to something more accurate like init on ... was sent successfully. Maybe you could even move this logging to Queue.

Reply:
I moved it to the Queue function. The logging for "init" will now show two traces like for "log". 

It's still using the _logger for the tracing however not the local logger like for "log". Was the use of _logger vs LocalLogger intentional for "init"? I guess it's because it's safe since there's no recursivity concern like there is for "log"?